File: test/language/parameterized_types/python/BytesParamTest.py
import unittest
import zserio

from testutils import getZserioApi

class BytesParamTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.api = getZserioApi(__file__, "parameterized_types.zs").bytes_param

    def testWriteRead(self):
        bytesData = bytearray([0xCA, 0xFE])
        bytesParam = self.api.BytesParam(bytesData, self.api.Parameterized(bytesData, 13))

        bitBuffer = zserio.serialize(bytesParam)
        readBytesParam = zserio.deserialize(self.api.BytesParam, bitBuffer)
        self.assertEqual(bytesParam, readBytesParam)

File: test/language/parameterized_types/java/parameterized_types/bytes_param/BytesParamTest.java
package parameterized_types.bytes_param;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import zserio.runtime.io.BitBuffer;
import zserio.runtime.io.SerializeUtil;

public class BytesParamTest
{
    @Test
    public void writeRead() throws IOException
    {
        final byte[] bytesData = new byte[] {(byte)0xCA, (byte)0xFE};
        final BytesParam bytesParam = new BytesParam(bytesData, new Parameterized(bytesData, 13));

        assertTrue(bytesParam.getBytesField() == bytesParam.getParameterizedField().getParam());

        final BitBuffer bitBuffer = SerializeUtil.serialize(bytesParam);
        final BytesParam readBytesParam = SerializeUtil.deserialize(BytesParam.class, bitBuffer);
        assertEquals(bytesParam, readBytesParam);
    }
}
